scripts/ci/dockerfiles/staking-miner/build.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+# Sample call:
+# $0 /path/to/folder_with_staking-miner_binary
+# This script replace the former dedicated staking-miner Dockerfile
+# and shows how to use the generic binary_injected.dockerfile
```

Comment:
This comment sounds wrong; staking-miner_builder.Dockerfile is still a dedicated Dockerfile that doesn't use binary_injected.dockerfile in any way.

Reply:
Actually, this is correct. The `binary_injected.dockerfile` only replaces the **injected** images.
The `staking-miner_builder.Dockerfile` is not an injected image: we build the binary and buil the image. So this is a different option to build an image. We don't use those much since they are pretty slow but when provided, those builder image allow a user building an image without anything installed/setup locally besides docker or podman.

I think we only have builder images atm for polkadot and staking-miner.

I will update the comment to be a little more specific.